src/linear_algebra/multilinear/basic.lean

```diff
@@ -676,6 +676,43 @@ def dom_dom_congr_linear_equiv {ι₁ ι₂} [decidable_eq ι₁] [decidable_eq
   .. (dom_dom_congr_equiv σ : multilinear_map A (λ i : ι₁, M₂) M₃ ≃+
         multilinear_map A (λ i : ι₂, M₂) M₃) }
 
+/-- The dependent version of `multilinear_map.dom_dom_congr_linear_equiv`. -/
+@[simps apply symm_apply]
+def dom_dom_congr_linear_equiv' {ι' : Type*} [decidable_eq ι'] (σ : ι ≃ ι') :
+  multilinear_map R M₁ M₂ ≃ₗ[R] multilinear_map R (λ i, M₁ (σ.symm i)) M₂ :=
+{ to_fun    := λ f,
+    { to_fun    := f ∘ (σ.Pi_congr_left' M₁).symm,
+      map_add'  := λ m i,
+        begin
+          rw ← σ.apply_symm_apply i,
+          intros x y,
+          simp only [comp_app, Pi_congr_left'_symm_update, f.map_add],
+        end,
+      map_smul' := λ m i c,
+        begin
+          rw ← σ.apply_symm_apply i,
+          intros x,
+          simp only [comp_app, Pi_congr_left'_symm_update, f.map_smul],
+        end, },
+  inv_fun   := λ f,
+    { to_fun    := f ∘ (σ.Pi_congr_left' M₁),
+      map_add'  := λ m i,
+        begin
+          rw ← σ.symm_apply_apply i,
+          intros x y,
+          simp only [comp_app, Pi_congr_left'_update, f.map_add],
+        end,
+      map_smul' := λ m i c,
+        begin
+          rw ← σ.symm_apply_apply i,
+          intros x,
+          simp only [comp_app, Pi_congr_left'_update, f.map_smul],
+        end, },
```

Comment:
It's nasty that you have to do this twice, but I can't see an easy way out of it. `Pi_congr_left'` isn't symmetric enough :(.